fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/EraseResourceDAO.java

```diff
@@ -0,0 +1,357 @@
+                stmt.setLong(1, logicalResourceId);
+                stmt.setInt(2, eraseDto.getVersion());
+                int count = stmt.executeUpdate();
+                LOG.fine(() -> "Count of resource_change_log deleted is " + count);
+            } catch (SQLException x) {
+                LOG.log(Level.SEVERE, RCL_DELETE, x);
+                throw translator.translate(x);
+            }
+
+            eraseRecord.setTotal(total);
+            eraseRecord.setStatus(ResourceEraseRecord.Status.VERSION);
+            return;
+        }
+
+        // Step 2: Delete from resource_change_log
+        final String RCL_DELETE =
+                "DELETE FROM RESOURCE_CHANGE_LOG"
+               + " WHERE RESOURCE_ID IN ("
+                   + " SELECT RESOURCE_ID"
+                   + " FROM " + eraseDto.getResourceType() + "_RESOURCES"
+                   + " WHERE LOGICAL_RESOURCE_ID = ?)";
+        try (PreparedStatement stmt = getConnection().prepareStatement(RCL_DELETE)) {
+            stmt.setLong(1, logicalResourceId);
+            int count = stmt.executeUpdate();
+            LOG.fine(() -> "Count of resource_change_log deleted is " + count);
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, RCL_DELETE, x);
+            throw translator.translate(x);
+        }
+
+        // Step 3: Delete All Versions from Resources Table
+        // Create the prepared statement to delete Resource Versions in chunks
+        // Implementation note: fetch must be the last part of the sub-select
+        final String DELETE_ALL_VERSIONS =
+                "DELETE FROM " + resourceType + "_RESOURCES" +
+                        "    WHERE LOGICAL_RESOURCE_ID = ?";
+        try (PreparedStatement stmt = getConnection().prepareStatement(DELETE_ALL_VERSIONS)) {
+            stmt.setLong(1, logicalResourceId);
+            total = stmt.executeUpdate();
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, DELETE_ALL_VERSIONS, x);
+            throw translator.translate(x);
+        }
+
+        // Step 4: Delete from parameters tables
+        deleteFromAllParametersTables(resourceType, logicalResourceId);
+
+        // Step 5: Delete from Logical Resources table
+        final String DELETE_LOGICAL_RESOURCE =
+                "DELETE FROM " + resourceType + "_LOGICAL_RESOURCES WHERE LOGICAL_RESOURCE_ID = ?";
+        try (PreparedStatement stmt = getConnection().prepareStatement(DELETE_LOGICAL_RESOURCE)) {
+            stmt.setLong(1, logicalResourceId);
+            int count = stmt.executeUpdate();
+            LOG.fine(() -> "Count of Resource_LR deleted is " + count);
+        } catch (SQLException x) {
+            LOG.log(Level.SEVERE, DELETE_LOGICAL_RESOURCE, x);
+            throw translator.translate(x);
+        }
+
+        // Step 6: Delete from Global Logical Resources
+        final String GLOBAL_DELETE_LOGICAL_RESOURCE =
```

Comment:
Minor nit here...you have the logical_resource_id value which is the PK for this table so would be the most efficient (and correct) to use for this deletion. But the lrid/resourcetypeid pair is also unique and in the end is logically the same.

Reply:
Got it... thanks - per convo, we can fix this up in the future.